### ISSUE TITLE
Auto-refresh and DAG double-trigger issues when toggling pause

### ISSUE THREAD

**Issue by q121212** [apache/airflow#35220]
### Apache Airflow version

2.7.2

### What happened

Problems that I discovered: 1) If DAG is on pause, then when you go into it and unpause it with the button on the left, the DAG trigger turns on, but auto-update does not turn on 2) if DAG is on pause, and you turn it on with the play button (on the right  ), then the dag is launched twice - as if once due to the inclusion of the dag and the second due to the activation of an already enabled dag

### What you think should happen instead

After unpause DAG button auto-update should be turned on (the same behavior like after clear task)

After push play button DAG should be triggered only one time

## Steps to Reproduce

1. Have a DAG on pause.
2. Navigate to the DAG's page in the Airflow UI.
3. Toggle the pause off using the trigger DAG button on the left.
4. Observe that the DAG is resumed but auto-refresh is not enabled.
5. Now, toggle the pause off using the play button on the right.
6. Observe that the DAG is triggered twice, seemingly once for the un-pausing and once for the activation of the already resumed DAG.

### Operating System

Linux from docker, and linux (rhel)

### Deployment

Virtualenv installation

### Deployment details

I think it's doesn't connected with deployment.
I have the same problems in 2 different envs and also with airflow more earlier versions 

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Taragolis** [comment]
> After push play button DAG should be triggered only one time

It is intended behaviour if in `schedule` defined cron expression or timetable.

**Comment by q121212** [comment]
> > After push play button DAG should be triggered only one time
> 
> It is intended behaviour if in `schedule` defined cron expression or timetable.

yes, my dags have cron defined

**Comment by Taragolis** [comment]
Yep in this case 1 would be for scheduling execution (it could be more if catchup=True) and another for manual

**Comment by q121212** [comment]
but this is strange, because there is no way to enable a dag that runs on a schedule so that it does not start additionally

**Comment by Taragolis** [comment]
That is always depends on requirements, if DAG intend to use as scheduled then you need to enable it (unpause), if you need manually run newly created DAG which intends to be run as scheduled then run will unpause and it run depends on its scheduler rule.

**Comment by Taragolis** [comment]
You might play with different [builtin Timetables](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/timetable.html#timetables), e.g. CronTriggerTimetable which would not spawn DagRun immediately on unpause

**Comment by jens-scheffler-bosch** [comment]
Yes, or if you define a start date in the future, then the activation of the DAG will not call the scheduler to run for the current scheduled period.
Turning a DAG "on" and running the immediate schedule is a feature I would say. If implicitly it is turned on when triggered is also a feature and then will result in two runs.
If you want to prevent a concurrency then there are also options in regards of parallel executions. Most probably  (at least in our environment) we have DAGs either scheduled or manual. But it might depend on your use case if you want to mix manual and scheduled runs in terms of status and reporting.

In terms of Auto-Refresh it was made in a way that if no DAG is running, Auto refresh is off. If you open the page and as long as any DAG runs, it will automatically refresh. It stops auto-refeshing if the last DAG is done. This is to prevent endless polling wasting webserver resources. Also Auto refresh is not active if the window is not in front.

Activating polling on un-pause only makes sense if the DAG has a schedule. I would judge it a small feature as improvement to enable refresh. If somebody wants to contribute, I would say this will be accepted. But I wont see this as "bug".

**Comment by q121212** [comment]
In our pipeline, we occasionally disable some DAGs to prevent them from executing. Another use case is during testing, based on the value of a variable, I decide which DAG should appear in the DAG list. It gets loaded but is inactive. I enable it and it triggers immediately, but since auto-refresh is not enabled, it's not apparent that it has been triggered and task processing has begun within it. Consequently, I hit play and then realize that it's already running. In this regard, I find such behavior to be non-intuitive for the user and hence consider it a bug. I understand it's not super critical, especially if you frequently interact with DAGs, you'll learn that Airflow operates in this manner. 
Overall, I agree that the problems described in this issue could also be viewed as suggestions for improving usability.

**Comment by William-Yi-Weng** [comment]
Is that expected that DAGs trigger immediately for last scheduled task even catchup=True once they are re-enabled since 2.7 for DeltaDataIntervalTimetable?

**Comment by jens-scheffler-bosch** [comment]
> Is that expected that DAGs trigger immediately for last scheduled task even catchup=False once they are re-enabled since 2.7 for DeltaDataIntervalTimetable?

Yes. The one run is immediately generated in the current time fits into one not covered current schedule. Additional (past) runs are generated with Catchup only if there are executions missing in the past.

**Comment by cedric-fauth** [comment]
> In our pipeline, we occasionally disable some DAGs to prevent them from executing. Another use case is during testing, based on the value of a variable, I decide which DAG should appear in the DAG list. It gets loaded but is inactive. I enable it and it triggers immediately, but since auto-refresh is not enabled, it's not apparent that it has been triggered and task processing has begun within it. Consequently, I hit play and then realize that it's already running. In this regard, I find such behavior to be non-intuitive for the user and hence consider it a bug.

Hi, we have the same issue with airflow. I understand this is expected but it would be a great enhancement if there was a feature to turn off the immediate-run behavior when unpausing a dag with `catchup=False`.

**Comment by potiuk** [comment]
> Hi, we have the same issue with airflow. I understand this is expected but it would be a great enhancement if there was a feature to turn off the immediate-run behavior when unpausing a dag with `catchup=False`.

Yep. If someone contributes that feature it would be nice. Maybe @cedric-fauth it's a good idea to contribute it ?

**Comment by cedric-fauth** [comment]
Unfortunately, I don't think I have the resources currently to get familiar with the source code and implement this.

**Comment by potiuk** [comment]
> Unfortunately, I don't think I have the resources currently to get familiar with the source code and implement this.

No worries - it's marged as good first issue, so hopefully someone will.

**Comment by Lzzz666** [comment]
May I ask whether this issue has been resolved?
Many companies are still using airflow v2.x. If still open, I'm happy to contribute it.

**Comment by jscheffl** [comment]
Probably it is open but the 2.x-line of Airflow is only maintained for critical security patches. Features and small bug fixes are only going into 3.x-line. Please upgrade. Only a few months to go an 2.x-line will be unmaintained.
